### DISCUSSION TITLE
One way communication with libjuice

### DISCUSSION THREAD

**Discussion by rcvangemert** [paullouisageneau/libjuice#213] [General]
still quietly working with libjuice and have my app streaming mpegts between devices. 

If I only need to connect to a remote device and receive a transport stream can I remove the second handshake?  I have confirmed that it does work but want to be sure there are no side effects.

FYI, it would have been nice to have rtp encoding/decoding as part of juice. but maybe I should have used libdatachannel!

**Comment by paullouisageneau** [comment]
I already answered you in https://github.com/paullouisageneau/libjuice/discussions/192#discussioncomment-5569615

There is no one way communication mode, you are just abusing the lack of timeout to get the description. You must forward both descriptions, otherwise the connection will typically fail when the first peer is behind a firewall or NAT.

If you don't want to carry out the signaling process, there is no point in using ICE, just send UDP packets from one peer to the other (you won't get firewall or NAT traversal capabilities, however).

**Comment by rcvangemert** [comment]
Thanks.
